src/main.rs

```diff
@@ -1,21 +1,34 @@
+//! Default Compute@Edge template program.
+
 use fastly::http::{HeaderValue, Method, StatusCode};
 use fastly::request::CacheOverride;
-use fastly::{downstream_request, Body, Error, Request, RequestExt, Response, ResponseExt};
+use fastly::{Body, Error, Request, RequestExt, Response, ResponseExt};
 use std::convert::TryFrom;
 
-const VALID_METHODS: [Method; 3] = [Method::HEAD, Method::GET, Method::POST];
+/// The name of a backend server associated with this service.
+///
+/// This should be changed to match the name of your own backend. See the the `Hosts` section of
+/// the Fastly WASM service UI for more information.
+const BACKEND_NAME: &str = "backend_name";
+
+/// The name of a second backend associated with this service.
+const OTHER_BACKEND_NAME: &str = "other_backend_name";
 
-/// Handle the downstream request from the client.
+/// The entrypoint for your application.
+///
+/// This function is triggered when your service receives a client request. It could be used to
+/// route based on the request properties (such as method or path), send the request to a backend,
+/// make completely new requests, and/or generate synthetic responses.
 ///
-/// This function accepts a Request<Body> and returns a Response<Body>. It could
-/// be used to route based on the request properties (such as method or path),
-/// send the request to a backend, make completely new requests and/or generate
-/// synthetic responses.
-fn handle_request(mut req: Request<Body>) -> Result<Response<Body>, Error> {
+/// If `main` returns an error a 500 error response will be delivered to the client.
+#[fastly::main]
+fn main(mut req: Request<Body>) -> Result<impl ResponseExt, Error> {
```

Comment:
:tada: 

We have a `fastly::main` macro now, which helps remove a lot of boilerplate :tada: 

https://docs.rs/fastly/0.3.0/fastly/attr.main.html

Reply:
I've updated the documentation for this function, pulling the relevant bits of `main` and `handle_request`. Happy to bikeshed :bike: